Manager/src/main/java/io/raspberrywallet/manager/bitcoin/Bitcoin.java

```diff
@@ -177,6 +186,15 @@ private void synchronizeWalletNonBlocking(final Wallet wallet, @Nullable KeyPara
         peerGroup.addPeerDiscovery(new DnsDiscovery(params));
         chain.addWallet(wallet);
         peerGroup.addWallet(wallet);
+        Runnable afterSynchronizationComplete = () -> {
+            try {
+                if (password != null)
+                    saveEncryptedWallet(password);
+                Logger.info("Wallet balance" + wallet.getBalance().toFriendlyString());
+            } catch (WalletNotInitialized | IOException e) {
```

Comment:
this method throws IOException anyways, maybe you can just throw it, instead of catching and printing stack trace

Reply:
Unfortunately, Java doesn't allow Lambdas to throw exceptions, it must be caught inside it.
A workaround is to use Kotlin ❤️ 🤣